carma-messenger-core/cpp_message/src/MobilityRequest_Message.cpp

```diff
@@ -0,0 +1,538 @@
+        uint8_t string_content_hostId[string_size];
+        for(size_t i=0;i<string_size;i++)
+        {
+            string_content_hostId[i]=plainMessage.header.sender_id[i];
+        }
+        message->value.choice.TestMessage00.header.hostStaticId.buf=string_content_hostId;
+        message->value.choice.TestMessage00.header.hostStaticId.size=string_size;
+        //convert target_id string to char array
+        string_size=plainMessage.header.recipient_id.size();
+        uint8_t string_content_targetId[string_size];
+        for(size_t i=0;i<string_size;i++)
+        {
+            string_content_targetId[i]=plainMessage.header.recipient_id[i];
+        }
+        message->value.choice.TestMessage00.header.targetStaticId.buf=string_content_targetId;
+        message->value.choice.TestMessage00.header.targetStaticId.size=string_size;
+        
+         //convert bsm_id string to char array
+        string_size=plainMessage.header.sender_bsm_id.size();
+        uint8_t string_content_BSMId[string_size];
+        for(size_t i=0;i<string_size;i++)
+        {
+            string_content_BSMId[i]=plainMessage.header.sender_bsm_id[i];
+        }
+        message->value.choice.TestMessage00.header.hostBSMId.buf=string_content_BSMId;
+        message->value.choice.TestMessage00.header.hostBSMId.size=string_size;
+        
+         //convert plan_id string to char array
+        string_size=plainMessage.header.plan_id.size();
+        uint8_t string_content_planId[string_size];
+        for(size_t i=0;i<string_size;i++)
+        {
+            string_content_planId[i]=plainMessage.header.plan_id[i];
+        }
+        message->value.choice.TestMessage00.header.planId.buf=string_content_planId;
+        message->value.choice.TestMessage00.header.planId.size=string_size;
+        //get timestamp and convert to char array
+        uint64_t time=plainMessage.header.timestamp;
+        std::string timestamp=std::to_string(time);
+        string_size=timestamp.size();
+        uint8_t string_content_timestamp[string_size];
+        for(size_t i=0;i<string_size;i++)
+        {
+            string_content_timestamp[i]=timestamp[i];
+        }
+        message->value.choice.TestMessage00.header.timestamp.buf=string_content_timestamp;
+        message->value.choice.TestMessage00.header.timestamp.size=string_size;
+        //strategy
+        string_size=plainMessage.strategy.size();
+        uint8_t string_content_strategy[string_size];
+        for(size_t i=0;i<string_size;i++)
+        {
+            string_content_strategy[i]=plainMessage.strategy[i];
+        }
+        message->value.choice.TestMessage00.body.strategy.buf=string_content_strategy;
+        message->value.choice.TestMessage00.body.strategy.size=string_size;
+
+        //plantype
+        MobilityPlanType_t type;
+            enum plan {UNKNOWN=0,CHANGE_LANE_LEFT=1,CHANGE_LANE_RIGHT=2,JOIN_PLATOON_AT_REAR=3,PLATOON_FOLLOWER_JOIN=4};
```

Comment:
Same comment re: this enum, if this really is the only place we've defined this enum it still might be worth moving it out into a header file just to make sure the encode and decode functions reference the same enum (in case we change the values, for example)

Reply:
I changed this, enum redefinition not needed.